streaming/base/dataset.py

```diff
@@ -765,6 +787,128 @@ def _resample_streams(self, epoch: int) -> Tuple[NDArray[np.int64], NDArray[np.i
         sample_ids = np.concatenate(sample_ids).astype(np.int64)
         return shuffle_units, sample_ids
 
+    def _generate_work_per_stream_batching(self, world: World, epoch: int,
```

Comment:
Do you know how much time does this method operation takes with today's standard dataset ?

Reply:
no i'm not sure. want me to measure? probably like a couple seconds more on large datasets, if at all.